### ISSUE TITLE
Fail fast approach / configuration

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#179]
I'd like to have a fail fast approach where I'd say in the configuration file that I want every rule to be automatically turned on with the default values. That would also include rules that are maybe turned off by default such as in https://github.com/arturbosch/detekt/pull/174.

The motivation behind this is that I want to fail fast and have every rule activated and then customize some rules to my needs. E.g. For instance changing the MaxLineLength to 100 instead of the default 120. Also with this approach the size of the configuration file would be reduced dramatically which is in favor when trying to keep them in sync across multiple projects.

Interesting in hearing opinions about this.

**Comment by arturbosch** [comment]
Hmm sounds interesting, basically a flag in the config like `failFast=true` which creates a `WrapperConfig` around the `DefaultConfig` which intercept all calls to `active` and return `true` ?

**Comment by vanniktech** [comment]
Yes something like that.

`failFast: true` where `warningThreshold` & `failThreshold` will be automatically set to 0. Weights can then be ignored and left untouched.

And if I want to adjust the `maxLineLength` value I just simply use this configuration file:

```yaml
autoCorrect: true

style:
  MaxLineLength:
    maxLineLength: 100
```

All rules are turned on by default and the value of `maxLineLength` is adjusted to 100. Now if I don't want to have the `CommentOverPrivateMethod` turned on I append:

```yaml
comments:
  CommentOverPrivateMethod:
    active: false
```

**Comment by arturbosch** [comment]
As of M12 you can override configuration values by passing two config paths to `--config`, so basically you can create a config file with everything turned on and with default thresholds and override some values with the second config. (`--config path/to/base,path/to/overridden`)

**Comment by vanniktech** [comment]
While this is nice it won't work well with different projects and version controlling. The base file would need to be checked into each project. 

I'm also happy to take over the implementation of the failfast feature if you approve of it.

**Comment by arturbosch** [comment]
Well it won't hurt, I would also use this option :)

Give it a try! Just my tipps: 

`Main.loadConfiguration`-function in cli builds the config object
There is a `CompositeConfig` class which allows you to combine two configs
